### PR TITLE
GEOMESA-1417 Allow custom quote characters in DelimitedTextConverter

### PR DESCRIPTION
**Pull request by anthonyccri**
* Added an optional field, 'options.quote', that controls what character
  is used for quoting

Signed-off-by: Anthony Fox <anthonyfox@ccri.com>

### CODE REVIEW

File: geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala [hunk 1]
```diff
@@ -46,20 +46,24 @@ class DelimitedTextConverterFactory extends AbstractSimpleFeatureConverterFactor
...
+    conf.getStringOpt("options.quote").foreach { q =>
+      baseFmt = baseFmt.withQuote(q.toCharArray()(0))
+    }
+
     new DelimitedTextConverter(baseFmt, sft, idBuilder, fields, userDataBuilder, opts, validating)
   }
 }
 
-class DelimitedOptions(var skipLines: Int = 0, var pipeSize: Int = 16 * 1024)
+class DelimitedOptions(var skipLines: Int = 0, var pipeSize: Int = 16 * 1024, var quote: String = "\"")
```
Comment (elahrvivaz) [inline geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala]: is this actually used, or could it be removed?

File: geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala [hunk 2]
```diff
@@ -46,20 +46,24 @@ class DelimitedTextConverterFactory extends AbstractSimpleFeatureConverterFactor
...
     val opts = {
-      import org.locationtech.geomesa.utils.conf.ConfConversions._
       val o = "options"
       val dOpts = new DelimitedOptions()
       conf.getIntOpt(s"$o.skip-lines").foreach(s => dOpts.skipLines = s)
       conf.getIntOpt(s"$o.pipe-size").foreach(p => dOpts.pipeSize = p)
       dOpts
     }
 
+    conf.getStringOpt("options.quote").foreach { q =>
```
Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala]: [maybe not important] if we wanted to disable quotes what would we do?

Comment (elahrvivaz) [inline geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala]: i believe most of the formats don't consider quotes - just the 'QUOTE*' one

**Comment by jahhulbert-ccri** [comment]
maybe in the future we should also allow a "escape" char so we can enable backslash escaping.

File: geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala]: cool

File: geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala (hunk 1 above)
Comment (jahhulbert-ccri) [inline geomesa-convert/geomesa-convert-text/src/main/scala/org/locationtech/geomesa/convert/text/DelimitedTextConverter.scala]: doesn't seem to be used

**Comment by elahrvivaz** [comment]
looks like MYSQL is the only format that sets an escape char (`\`), so yeah exposing that would be good

**Comment by jahhulbert-ccri** [comment]
@anthonyccri can you add an argument for escaping? maybe ```options.escape```

**Comment by jahhulbert-ccri** [comment]
@elahrvivaz if you are ok with my additions we're good to go

**Comment by elahrvivaz** [comment]
looks good - merging

**Pull request merged**